src/Lumi/Components2/Text.purs

```diff
@@ -0,0 +1,194 @@
+module Lumi.Components2.Text where
+
+import Prelude
+
+import Data.Lens (Setter')
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, LumiProps, PropsModifier, lumiComponent, propsModifier)
+import Lumi.Styles as S
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic (JSX, ReactComponent)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Hooks as Hooks
+
+-- Text
+
+data Text = Text
+
+type TextProps =
+  ( component :: Text
+  , element :: TextElement
+  , content :: String
+  )
+
+type TextElement = ReactComponent { children :: Array JSX, className :: String }
+
+text :: LumiComponent TextProps
+text =
+  unsafePerformEffect $ lumiComponent "Text" defaults \props -> Hooks.do
+    theme <- useTheme
+    pure $
+      E.element props.element
+        { children: [ R.text props.content ]
+        , className: props.className
+        , css: defaultTextStyle <> props.css theme
+        }
+  where
+    defaults :: Record TextProps
+    defaults =
+      { component: Text
+      , element: R.span'
+      , content: ""
+      }
+
+    defaultTextStyle =
+      S.css
+        { fontSize: S.str "14px"
+        , lineHeight: S.str "17px"
+        , margin: S.str "0"
+        , padding: S.str "0"
+        }
+
+type TextModifier = forall r. PropsModifier (component :: Text, element :: TextElement | r)
+
+_body :: TextModifier
+_body = propsModifier _{ element = R.span' :: TextElement }
+
+_strong :: TextModifier
+_strong = propsModifier _{ element = R.strong' :: TextElement }
+
+_emphasis :: TextModifier
+_emphasis = propsModifier _{ element = R.em' :: TextElement }
+
+_subtext :: TextModifier
+_subtext =
+  propsModifier _{ element = R.small' :: TextElement }
+  <<< S.style_
+        ( S.css
+            { fontSize: S.str "12px"
+            , lineHeight: S.str "14px"
+            , marginBottom: S.str "6px"
+            }
+        )
+
+_muted :: TextModifier
+_muted =
+  S.style \(LumiTheme { colors }) ->
+    S.css { color: S.color colors.black1 }
+
+-- Paragraph
+
+type ParagraphProps =
+  ( component :: Text
+  , element :: TextElement
+  , content :: Array JSX
+  )
+
+paragraph_ :: LumiComponent TextProps
+paragraph_ = paragraph <<< renderInnerText
+  where
+    renderInnerText :: Setter' (LumiProps ParagraphProps) (LumiProps TextProps)
```

Comment:
Since this isn't being used with any lens functions, I feel like it might be easier to read if you expanded this `Setter'` type synonym.